ArchiSteamFarm/Statistics.cs

```diff
@@ -357,16 +369,16 @@ internal sealed class Statistics : IDisposable {
 				return false;
 			}
 
-			(Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), Dictionary<ulong, uint>> fullState, Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), Dictionary<ulong, uint>> tradableState) = Trading.GetDividedInventoryState(ourInventory);
+			(Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), Dictionary<ulong, uint>> ourFullState, Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), Dictionary<ulong, uint>> ourTradableState) = Trading.GetDividedInventoryState(ourInventory);
 
-			if (Trading.IsEmptyForMatching(fullState, tradableState)) {
+			if (Trading.IsEmptyForMatching(ourFullState, ourTradableState)) {
 				// User doesn't have any more dupes in the inventory
-				Bot.ArchiLogger.LogGenericTrace(string.Format(Strings.ErrorIsEmpty, nameof(fullState) + " || " + nameof(tradableState)));
+				Bot.ArchiLogger.LogGenericTrace(string.Format(Strings.ErrorIsEmpty, nameof(ourFullState) + " || " + nameof(ourTradableState)));
 
 				return false;
 			}
 
-			ImmutableHashSet<ListedUser> listedUsers = await GetListedUsers().ConfigureAwait(false);
+			ImmutableHashSet<ListedUser> listedUsers = await GetListedUsers(MatchEverything).ConfigureAwait(false);
```

Comment:
Why did you decide to complicate the main function and send two requests? You should send one request, order the loop by listedUser.MatchEverything and adapt the logic appropriately. This made the PR worse than before.

Reply:
Guess I'm too slow at writing. Will revert that.

Reply:
Or you may wait a while more for me to finish reply to your latest message.